bootstrap/bootstrap.go

```diff
@@ -1756,15 +1756,16 @@ func (b *Bootstrap) writeBatchScript(cmd string) (string, error) {
 	}
 	defer scriptFile.Close()
 
-	var scriptContents = "@echo off\n"
+	var scriptContents = []string{"@echo off"}
 
 	for _, line := range strings.Split(cmd, "\n") {
 		if line != "" {
-			scriptContents += line + "\n" + "if %errorlevel% neq 0 exit /b %errorlevel%\n"
+			scriptContents = append(scriptContents, "call " + line)
```

Comment:
@aknarayan I’m not very familiar with the Batch shell, are there occasions where this would not be correct? Should this be looking for lines that have a "command" like prelude e.g. "thing.ext"?

Reply:
that's correct, but if we can safely assume that `line` will be the name of another script intended to be called in a command step then the use of `call` here should be fine. i don't know how the agent parses this, but for example, if a command step has an attribute that looks like this:
```
commands:
	- a.bat
	- b.bat
```
and when parsed, `line` will only ever take the values `a.bat` and `b.bat`, then the use of `call` here is correct.

Reply:
it might also be better to leave the `errorlevel` check in there as well; batch sometimes behaves weirdly with error levels (more detail [here](https://waynes-world-it.blogspot.com/2008/08/difference-between-bat-and-cmd.html)) and i'm not too familiar with it either, but i don't think it can hurt to leave the `errorlevel` check in.

Reply:
> if we can safely assume that `line` will be the name of another script intended to be called in a command step

That’s what I’m wondering, is there other Batch shell valid syntax that you could include in the "command" field of a step that would have worked before, like variable assignment perhaps?

Reply:
yeah, potentially - in that case it would be safer to use the `call` keyword iff `line` is of the format `thing.ext`.